src/Driver/Compiler.php

```
@@ -475,15 +475,10 @@ protected function condition(QueryParameters $params, Quoter $q, array $context)
 
         $placeholder = '?';
         if ($value->isArray()) {
-            if ($operator === '=') {
-                $operator = 'IN';
-            } elseif ($operator === '!=') {
-                $operator = 'NOT IN';
-            }
+            return $this->arrayToInOperator($params, $q, $value->getValue(), $operator === 'IN' || $operator === '=');
```

It's better to use match-case there and throw an exception if we get an unsupported operator

---

I'll do it